### DISCUSSION TITLE
mergerfs.percent-full-mover script - 13: [: Verw: unexpected operator

### DISCUSSION THREAD

**Discussion by ghost** [trapexit/mergerfs#1314] [Q&A] [answered]
Hey all,

I am trying to run the script from [here ](https://raw.githubusercontent.com/trapexit/mergerfs/latest-release/tools/mergerfs.percent-full-mover) but have problems running it.. I prob. just dont understand arguments for bash scripts.

The command I try is 
sh mergerfs.sh /srv/dev-disk-by(etc to my cache) /srv/mergerfs/SnapraidSlow 5

I am trying to test it with just one file so I'd like 5%. I get an error though: 
mergerfs.sh: 13: [: Verw: unexpected operator

I think my arguments for the FS of the cache drive are incorrect.. Maybe someone can point me in the right direction here. Thanks alot :)

**Comment by trapexit** [comment]
What shell are you using?

**Reply by ghost** [reply]
Wow, you are quick.

Bash (on  Debian 6.1.76-1 OpenMediaVault 7)

**Reply by trapexit** [reply]
Are you sure? 

`ls -lh $(which sh)`

**Reply by ghost** [reply]
Apparently not.. sorry
/usr/bin/sh -> dash

**Reply by trapexit** [reply]
How did you download the script?

```
$ wget  https://raw.githubusercontent.com/trapexit/mergerfs/latest-release/tools/mergerfs.percent-full-mover
$ chmod +x mergerfs.percent-full-mover
$ ./mergerfs.percent-full-mover
usage: mergerfs.percent-full-mover <cache-fs> <backing-pool> <percentage>
$ ./mergerfs.percent-full-mover /cache-fs /mergerfs 5
```

**Reply by ghost** [reply]
I used nano to copy/paste the script and then did a chmod a+x. 
Tried you variant and get:
```
root@NAS-Breece2:~# ./mergerfs.percent-full-mover /srv/dev-disk-by-uuid-fafc3b71-0e57-42e5-a497-1067a8bac008 /srv/mergerfs/SnapraidSlow 5
./mergerfs.percent-full-mover: 13: [: Verw: unexpected operator


```

**Reply by trapexit** [reply]
Can you please use wget or curl to download it? Or git? You're clearly messing something up with the script in your copy and paste. I link to the file to download so you aren't doing that.

**Reply by ghost** [reply]
I meant to say, that I tried your steps like you described them:
```
$ wget  https://raw.githubusercontent.com/trapexit/mergerfs/latest-release/tools/mergerfs.percent-full-mover
$ chmod +x mergerfs.percent-full-mover
```

I then tried to use the new script and got:

```
root@NAS-Breece2:~# ./mergerfs.percent-full-mover /srv/dev-disk-by-uuid-fafc3b71-0e57-42e5-a497-1067a8bac008 /srv/mergerfs/SnapraidSlow 5
./mergerfs.percent-full-mover: 13: [: Verw: unexpected operator
```

I am sorry for the misunderstanding.

**Reply by trapexit** [reply]
I don't know what to tell you. Something is messed up with either the way you've gotten the script or your system. I have OMV 7 installed and it works with dash just fine. Try bash?

**Reply by ghost** [reply]
First let me say thank you for your quick help. Its very much appreciated.
With bash I get a line 13 [: Too many arguments.. 
I got the script via wget and the system is a fresh install from today.

I did some testing though, and maybe that helps:
I changed
`#while [ $(df --output=pcent "${CACHE}" | grep -v Use | cut -d'%' -f1) -gt ${PERCENTAGE} ]
`
to a simple "while true" just to see if the rest of the script would work and it does.. So there is something wrong, somehow just on my system, with the while conditions part of the loop...

**Reply by trapexit** [reply]
I really don't know. Have you tried running the command seen in the while loop yourself on the command line? Have you tried different arguments?

I have a OMV 7 system, freshly installed this morning, I created your exactly directly layout and it works fine. I created /tmp/foo and /tmp/bar and ran ./mergerfs.percent-full-mover /tmp/foo /tmp/bar 5 and that too works fine.

**Reply by trapexit** [reply]
Can you run `sh -x ./mergerfs.percent-full-mover /srv/dev-disk-by-uuid-fafc3b71-0e57-42e5-a497-1067a8bac008 /srv/mergerfs/SnapraidSlow 5`

**Reply by ghost** [reply, marked as answer]
I've found the error and it was something specific to "my" system. I checked the output of the while conditions,yes:
`root@NAS-Breece2:~# df --output=pcent /srv/dev-disk-by-uuid-fafc3b71-0e57-42e5-a                                                                        497-1067a8bac008
Verw%
   1%
`

Most of my Debian appears to be in english, but SOME things are as per my location in GERMAN. So it does not result in USE% but in Verw% (Verwendung is Usage in German)... Ofc. grep cant work with that...

With the corrected line
`while [ $(df --output=pcent "${CACHE}" | grep -v Verw | cut -d'%' -f1) -gt ${PERCENTAGE} ]
`

everything works as its supposed to. I hope this might help others. Thank you very much for the assistance!

**Reply by trapexit** [reply]
I had wondered what "Verw" was doing there. I have German family but never picked up the language. The fact other things were English threw me off. Glad you got it working.

**Reply by ghost** [reply]
Threw me off as well because almost everything is in english.. Oh well, now it works.

**Reply by ghost** [reply]
Ok, I have to reply again:
When I SSH into my OMV7 machine df outputs in german, but now that I added the script to the crontab via OMV gui it changed back to english so I had to change the script back to where it was...

**Reply by trapexit** [reply]
You might want to force the language via LANG and/or LC_ALL variables. Perhaps I should add those to the script (or just rewrite it in Python or something.)

**Reply by ghost** [reply]
Yeah, thats what I did (change the locale). Adding it to the script (atleast as a comment) might be helpful for others :)
